onnxruntime/python/onnxruntime_pybind_mlvalue.h

```diff
@@ -22,7 +22,8 @@ namespace py = pybind11;
 
 int OnnxRuntimeTensorToNumpyType(const DataTypeImpl* tensor_type);
 
-void CreateGenericMLValue(AllocatorPtr alloc, const std::string& name_input, py::object& value, OrtValue* p_mlvalue);
+void CreateGenericMLValue(const onnxruntime::InferenceSession* sess, AllocatorPtr alloc, const std::string& name_input,
+                          py::object& value, OrtValue* p_mlvalue);
```

Comment:
Not quite understanding why the lookup of the type from the session would be within CreateGenericMLValue.

Reply:
I need to get hold of the data type of the input sequence from the model. The only way to get this from the session is to call GetInputDefList() which returns a vector requiring a linear search of all the inputs for a specific name. Given that this is required for sequence type inputs only I didn't want to pay this cost for regular tensor types, which is why this type determination takes place in CreateSequenceOfTensors(). I can pass InputDefList* instead of the session ptr. May be that'll look better?

Reply:
New commit uses InputDefList.